scenarios/fork/requires-python-full.toml

```diff
@@ -0,0 +1,22 @@
+name = "fork-requires-python-full"
+description = '''
+This tests that a `Requires-Python` specifier will result in the
+exclusion of dependency specifications that cannot possibly satisfy it.
+
+In particular, this is tested via the `python_full_version` marker
+instead of the more common `python_version` marker.
+'''
+
+[resolver_options]
+universal = true
+
+[expected]
+satisfiable = false
+
+[root]
+requires_python = ">=3.10"
+requires = [
+  "a==1.0.0 ; python_full_version == '3.9b1'",
```

Comment:
Should we be including the pre-release here? Seems like it complicates the scenario and we should have coverage for that separately.

Reply:
(I found it surprising that it was not the same as the `requires-python` test)

Reply:
The reason why I used it was because `python_full_version` specifically allows it. So it seemed nice to test that aspect of it. But I can see how it's potentially testing more than one thing. I can split this apart.

Reply:
Okey dokey, fixed!

Reply:
Ahh that makes sense. This seems better anyway. Thanks!